drivers/sensor/melexis/MLX90394/mlx90394.c

```diff
@@ -0,0 +1,601 @@
+/*
+ * Copyright (c) 2024 Florian Weber <Florian.Weber@live.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT melexis_mlx90394
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor_data_types.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+#include "mlx90394.h"
+#include "mlx90394_reg.h"
+
+LOG_MODULE_REGISTER(MLX90394, CONFIG_SENSOR_LOG_LEVEL);
+
+static void mlx90394_convert_magn(enum mlx90394_reg_config_val config, struct sensor_value *val,
+				  uint8_t sample_l, uint8_t sample_h)
+{
+	int64_t scale;
+
+	if (config == MLX90394_CTRL2_CONFIG_HIGH_SENSITIVITY_LOW_NOISE) {
+		scale = MLX90394_HIGH_SENSITIVITY_MICRO_GAUSS_PER_BIT;
+	} else {
+		scale = MLX90394_HIGH_RANGE_MICRO_GAUSS_PER_BIT;
+	}
+	int64_t conv_val = (int16_t)((uint16_t)sample_l | (uint16_t)(sample_h << 8)) * scale;
```

Comment:
Declare all variables at the top of the function.

Reply:
accepted Thanks for the detailed review :)